packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart

```diff
@@ -0,0 +1,551 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:flutter/animation.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/physics.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+
+import 'framework.dart';
+import 'notification_listener.dart';
+import 'scroll_context.dart';
+import 'scroll_controller.dart';
+import 'scroll_metrics.dart';
+import 'scroll_notification.dart';
+import 'scroll_physics.dart';
+import 'scroll_position.dart';
+import 'scroll_position_with_single_context.dart';
+import 'scrollable.dart';
+
+/// A controller for scroll views whose items have the same size.
+///
+/// Similar to a standard [ScrollController] but with the added convenience
+/// mechanisms to read and go to item indices rather than a raw pixel scroll
+/// offset.
+///
+/// See also:
+///
+///  * [ListWheelScrollView], a scrollable view widget with fixed size items
+///    that this widget controls.
+///  * [FixedExtentMetrics], the `metrics` property exposed by
+///    [ScrollNotification] from [ListWheelScrollView] which can be used
+///    to listen to the current item index on a push basis rather than polling
+///    the [FixedExtentScrollController].
+class FixedExtentScrollController extends ScrollController {
+  /// Creates a [FixedExtentScrollController].
+  FixedExtentScrollController({
+    this.initialItem: 0,
+  }) : assert(initialItem != null);
+
+  /// The page to show when first creating the scroll view.
+  ///
+  /// Defaults to 0 and must not be null.
+  final int initialItem;
+
+  /// The currently selected item index that's closest to the center of the viewport.
+  ///
+  /// There are circumstances that this [FixedExtentScrollController] can't know
+  /// the current item. Reading [selectedItem] will throw an [AssertionError] in
+  /// the following cases:
+  ///
+  /// 1. No scroll view is currently using this [FixedExtentScrollController].
+  /// 2. More than one scroll views using the same [FixedExtentScrollController].
+  ///
+  /// The [hasClients] property can be used to check if a scroll view is
+  /// attached prior to accessing [selectedItem].
+  int get selectedItem {
```

Comment:
you go to great lengths to explain all this here, but animateToItem has the same limitations but no comment is made of it.

Reply:
also jumpToItem